Libraries/Components/Button.js

```diff
@@ -299,7 +308,7 @@ class Button extends React.Component<ButtonProps> {
         nextFocusRight={nextFocusRight}
         nextFocusUp={nextFocusUp}
         testID={testID}
-        disabled={disabled}
+        disabled={disabled != null ? disabled : accessibilityState?.disabled} //This will sync accessibilityState={disabled} with disabled prop.
```

Comment:
I think this should be done on line 286? Otherwise we're missing the logic in that if statement if accessibilityState.disabled is true

Reply:
I was a bit confused on this. For now it's not appyling the `disable style` if only accessibilityState?.disabled is passed.

Reply:
I think this needs to be passed here and on line 286 so it applies style and disables the button aswell if accessibilityState.disabled is true

Reply:
Yea I think we should treat the button as disabled if accessibilityState.disabled is true in all manners --- behaviour and styling

Reply:
> Yea I think we should treat the button as disabled if accessibilityState.disabled is true in all manners --- behaviour and styling

I think I agree with you -I think if accessibilityState.disabled=true and disabled=false though, disabled should take precedence. If disabled is undefined, then I agree accessibilityState.disabled=true should disable the component.

Reply:
@lunaleaps @kacieb thank you for the explanation. I have updated the accessibilityState.disabled=true to disable styling aswell.